### PR TITLE
Render results without reloading page

### PR DESCRIPTION
**Pull request by dnaranjo89**
**Note**: The first time a draw is tossed, the is still a page reload (the change between ```new_draw.html``` and ```display_draw.html```)

### CODE REVIEW

File: static/js/coin.js [hunk 1]
```diff
@@ -1,45 +1,96 @@
...
+     *     COIN CLASS DEFINITION
+     ********************************/
+    var Coin = function (element, options){
+        this.init(element, options);
+    };
+
+    Coin.prototype = {
+
+        constructor: Coin,
+
```
Comment (Mariocj89) [inline static/js/coin.js]: why so many spaces?

File: static/js/coin.js [hunk 2]
```diff
@@ -1,45 +1,96 @@
...
+        animation_frame_counter: 0,
+
+        result: null,
+
+        init: function (element, options) {
+            this.$element = $(element);
+            this.options = $.extend({}, defaults, options);
+
+            this.cache_images();
```
Comment (Mariocj89) [inline static/js/coin.js]: maybe we should do this async? Not sure though

File: static/js/coin.js [hunk 3]
```diff
@@ -1,45 +1,96 @@
...
+            this.image_store[2] = new Image();
+            this.image_store[2].src = this.options.base_url + "tail.png";
+            this.image_store[3] = new Image();
+            this.image_store[3].src = this.options.base_url + "head_dist.png";
+            this.image_store[4] = new Image();
+            this.image_store[4].src = this.options.base_url + "dist.png";
+        },
+
+        // Store the positions of the blurry images in order to show the animation
+        blurry_images_array: [3, 4, 1, 4],
```
Comment (Mariocj89) [inline static/js/coin.js]: 3, 4 1 4?

File: static/js/coin.js [hunk 4]
```diff
@@ -1,45 +1,96 @@
...
+
+        animate: function(){
+            var that = this;
+            function animate_loop(){
+                var image_frame = (that.animation_frame_counter) % 4;
```
Comment (Mariocj89) [inline static/js/coin.js]: why the parenthesis? is 4 the len of blurry_images array?

File: static/js/coin.js [hunk 5]
```diff
@@ -1,45 +1,96 @@
...
+
+    };
+
+    /*********************************
+     *   COIN PLUGIN DEFINITION
+     ********************************/
+    $.fn.coin = function (option, param) {
+        return this.each(function () {
+            var $this = $(this)
+                , data = $this.data('plugin_' + pluginName)
```
Comment (Mariocj89) [inline static/js/coin.js]: is it that hard to write bar multiple times?

File: static/js/coin.js (hunk 2 above)
Comment (dnaranjo89) [inline static/js/coin.js]: Could be. In normal sotuations there will be time enough for them to load. if not, the first time the animation runs it won't be very smooth, that's all. So yes we can do it async

File: static/js/coin.js (hunk 3 above)
Comment (dnaranjo89) [inline static/js/coin.js]: This is the images sequence to be shown when the coin is spinning

**Pull request merged**